Sources/Basic/Path.swift

```diff
@@ -342,132 +209,244 @@ private struct PathImpl {
     private let string: String
     
     /// Private function that returns the directory part of the stored path
-    /// string (relying on the fact that it has been normalized).  Returns nil
-    /// if there is no directory part (which is the case iff there is no path
-    /// separator).
-    private var dirname: String? {
+    /// string (relying on the fact that it has been normalized).  Returns an
+    /// empty string if there is no directory part (which is the case if and
+    /// only if there is no path separator).
+    private var dirname: String {
+        // FIXME: This method seems too complicated; it should be simplified,
+        //        if possible, and certainly optimized (using UTF8View).
         let chars = string.characters
-        guard let idx = chars.rindex(of: PathSeparatorCharacter) else {
-            return nil
+        // Find the last path separator.
+        guard let idx = chars.rindex(of: pathSeparatorCharacter) else {
+            // No path separators, so the directory name is empty.
+            return ""
```

Comment:
Would it make sense for dirname to return `"."` here? That seems like it might be consistent with the handling of `RelativePath("")`.